reference/test/reorder/mc64.cpp

```diff
@@ -0,0 +1,156 @@
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#include <ginkgo/core/reorder/mc64.hpp>
+
+
+#include <algorithm>
+#include <fstream>
+#include <memory>
+
+
+#include <gtest/gtest.h>
+
+
+#include <ginkgo/core/base/composition.hpp>
+#include <ginkgo/core/base/executor.hpp>
+#include <ginkgo/core/matrix/csr.hpp>
+#include <ginkgo/core/matrix/dense.hpp>
+#include <ginkgo/core/matrix/permutation.hpp>
+#include <ginkgo/core/matrix/scaled_permutation.hpp>
+#include <ginkgo/core/matrix/sparsity_csr.hpp>
+
+
+#include "core/test/utils.hpp"
+#include "core/test/utils/assertions.hpp"
+
+
+namespace {
+
+
+template <typename ValueIndexType>
+class Mc64 : public ::testing::Test {
+protected:
+    using v_type =
+        typename std::tuple_element<0, decltype(ValueIndexType())>::type;
+    using i_type =
+        typename std::tuple_element<1, decltype(ValueIndexType())>::type;
+    using real_type = gko::remove_complex<v_type>;
+    using reorder_type = gko::experimental::reorder::Mc64<v_type, i_type>;
+    using perm_type = gko::matrix::ScaledPermutation<v_type, i_type>;
+    using result_type = gko::Composition<v_type>;
+    using Mtx = gko::matrix::Dense<v_type>;
+    using CsrMtx = gko::matrix::Csr<v_type, i_type>;
+    Mc64()
+        : exec(gko::ReferenceExecutor::create()),
+          mc64_factory(reorder_type::build().on(exec)),
+          // clang-format off
+          id3_mtx(gko::initialize<CsrMtx>(
+              {{1.0, 0.0, 0.0}, 
+              {0.0, 1.0, 0.0}, 
+              {0.0, 0.0, 1.0}}, exec)),
+          not_id3_mtx(gko::initialize<CsrMtx>(
+              {{1.0, 0.0, 2.0}, 
+              {0.0, 1.0, 0.0}, 
+              {2.0, 0.0, 1.0}}, exec)),
+          // clang-format on
+          reorder_op(mc64_factory->generate(id3_mtx))
+    {}
+
+    std::pair<std::shared_ptr<const perm_type>,
+              std::shared_ptr<const perm_type>>
+    unpack(const result_type* result)
+    {
+        GKO_ASSERT_EQ(result->get_operators().size(), 2);
+        return std::make_pair(gko::as<perm_type>(result->get_operators()[0]),
+                              gko::as<perm_type>(result->get_operators()[1]));
+    }
+
+    void assert_correct_permutation(const result_type* mc64)
+    {
+        auto perm = unpack(mc64).first->get_const_permutation();
+
+        ASSERT_EQ(perm[0], 0);
+        ASSERT_EQ(perm[1], 1);
+        ASSERT_EQ(perm[2], 2);
+    }
+
+    std::shared_ptr<const gko::Executor> exec;
+    std::shared_ptr<CsrMtx> id3_mtx;
+    std::shared_ptr<CsrMtx> not_id3_mtx;
```

Comment:
this is not used. I guess at some point there was a test with that (or at least intended)